### PR TITLE
Update Maintainer's Guide

### PR DESCRIPTION
**Pull request by misscoded**
###  Summary

The last run through the release process brought to light some deficiencies in our Maintainer's Guide. This PR aims to shore up those deficiencies and remove references to `lerna`. 

Removing `lerna` all together (`package.json`) is out of scope for this PR, but should be revisited.

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: .github/maintainers_guide.md
Comment (seratch) [inline .github/maintainers_guide.md:47]: Before starting the release processes, we can have the step to check the latest status of GitHub milestones that we are going to ship as unintentionally some still may have open issues. 

If everything is ready for release, we can start the process. Otherwise, we can discuss whether we should move the remaining tasks to future releases or hold off the release until they're resolved.

**Review by seratch — APPROVED** [review]
Great work 👍

**Review by srajiang — COMMENTED** [review]
@misscoded My comments are mostly questions for clarification. The changes you've made to the doc are making sense to me (with the current context I have have!)

File: .github/maintainers_guide.md
Comment (misscoded) [reply .github/maintainers_guide.md:6]: Great question! The vulnerabilities look to be all associated with `lerna`. Since we're moving towards removing `lerna` all together, I don't believe it's worth fixing it beforehand, but if any if any of the other maintainers disagree on that point, we should definitely create an issue to address it.

File: .github/maintainers_guide.md
Comment (misscoded) [reply .github/maintainers_guide.md:11]: 🤦🏼‍♀️ You're totally right. I'll clean up the language to be clearer.

File: .github/maintainers_guide.md
Comment (misscoded) [reply .github/maintainers_guide.md:26]: I took a look because I wanted to make sure I didn't overlook anything, but `lerna` is only referenced at the root level `package.json`: none of the individual packages rely on it. I would have removed it entirely, but Steve hypothesized that we might cause CI errors, so I've omitted that from this PR.

Are you running these commands from the root level?

File: .github/maintainers_guide.md
Comment (misscoded) [reply .github/maintainers_guide.md:47]: Ah, great point! I'll add it in.

File: .github/maintainers_guide.md
Comment (misscoded) [reply .github/maintainers_guide.md:51]: Love this suggestion! Will add it.

File: .github/maintainers_guide.md
Comment (misscoded) [reply .github/maintainers_guide.md:11]: I just now realized I misread your comment! I think you had it right: when running tests, you need to navigate to the individual package (ie, `web-api`, `events-api`, etc.).

Either way, this can do with more clarity, I'm sure, so will still tweak it a bit.

File: .github/maintainers_guide.md
Comment (mwbrooks) [inline .github/maintainers_guide.md:186]:
```suggestion
Triaging is the process of investigating new issues, assigning them an appropriate label, and responding to the submitting developer. An issue should have **one** of the following labels applied: `bug`, `enhancement`, `question`, `needs info`, `docs`, `tests`, or `discussion`.
```

**Review by mwbrooks — APPROVED** [review]
Big improvement! 🌮

**Pull request merged**
